src/main/java/bio/terra/service/dataset/flight/upgrade/predictableFileIds/ConvertToPredictableFileIdsBqDropStageTableStep.java

```diff
@@ -0,0 +1,43 @@
+package bio.terra.service.dataset.flight.upgrade.predictableFileIds;
+
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.tabulardata.google.bigquery.BigQueryDatasetPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConvertToPredictableFileIdsBqDropStageTableStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(ConvertToPredictableFileIdsBqDropStageTableStep.class);
+
+  private final UUID datasetId;
+  private final DatasetService datasetService;
+  private final BigQueryDatasetPdao bigQueryDatasetPdao;
+
+  public ConvertToPredictableFileIdsBqDropStageTableStep(
+      UUID datasetId, DatasetService datasetService, BigQueryDatasetPdao bigQueryDatasetPdao) {
+    this.datasetId = datasetId;
+    this.datasetService = datasetService;
+    this.bigQueryDatasetPdao = bigQueryDatasetPdao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    Dataset dataset = datasetService.retrieve(datasetId);
+
+    // Drop the table
+    bigQueryDatasetPdao.deleteStagingFileIdMappingTable(dataset);
+
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    return StepResult.getStepResultSuccess();
```

Comment:
Checking that we don't want to restore the staging table in an undo, because if we're unwinding a flight we'll drop the table in the step that creates the table, right?

Reply:
that's exactly right!